types/node/index.d.ts

```diff
@@ -87,6 +87,16 @@ interface SharedArrayBuffer {
     slice(begin?: number, end?: number): SharedArrayBuffer;
 }
 
+interface PromiseSettlement<T> {
+    status: 'rejected' | 'fulfilled';
+    reason?: any;
+    value?: T;
+}
+
+interface PromiseConstructor {
+    allSettled<R>(iterable: Iterable<R | PromiseLike<R>>): Promise<Array<PromiseSettlement<R>>>;
```

Comment:
I’d rather we do https://github.com/microsoft/TypeScript/pull/34065.

Reply:
Agree, I actually posted a comment about this in another issue (which I can't find anymore), I will remove this for now and wait for the lib to avoid future conflicts.